src/main/src/main/java/org/geoserver/catalog/StyleType.java

```diff
@@ -10,6 +10,7 @@ public enum StyleType {
     LINE("Line"), // Line Feature Type
     POLYGON("Polygon"), // Polygon Feature Type
     RASTER("Raster"), // Coverage Type
+    GRID("Graticule"),
```

Comment:
I would not touch core for the benefit of a plugin. What about just providing a style example in the documentation? I think users will want to heavily customize the basic style anyways.

The template style can also be simplified significantly by using the env variables and some more advanced expressions. For example, if the process for extracting the label points accepted a filter, we could make it skip the origin graticule lines that are not of interest like this (mind hand-written and not tested):

```xml
   <ogc:PropertyIsLowerThanOrEqualTo>
          <ogc:PropertyName>level</ogc:PropertyName>
          <ogc:Function name="Categorize">
              <ogc:Function name="env"><ogc:Literal>wms_scale_denominator</ogc:Literal><ogc:Function>
              <ogc:Literal>0</ogc:Literal>
              <ogc:Literal>4000000</ogc:Literal>
              <ogc:Literal>1</ogc:Literal>
              <ogc:Literal>9000000</ogc:Literal>
              <ogc:Literal>2</ogc:Literal>
              <ogc:Literal>34000000</ogc:Literal>
              <ogc:Literal>3</ogc:Literal>
        </ogc:Function>              
   <ogc:PropertyIsLowerThanOrEqualTo>
```

The same approache could be applied to the lines themselves. The filter is basically static for the single request, and would evaluate in "level < staticValue" during style simplification.

For the text symbolizers, I'm thinking one would have to repeat over and over the font definition and its details. Maybe those could become data driven, and then we could have a single text symbolizer instead?
E.g., the anchor point x and y could become attributes, and the offsets could be -1,0,1 that one multiplies to the actual amount of offset desired, e.g.

```xml
<sld:DisplacementX><ogc:Mul><ogc:PropertyName>offsetX</ogc:PropertyName><ogc:Literal>5</ogc:Literal></sld:DisplacementX>
```

This would drastically reduce the size of the style, IMHO, and help users concentrate more on how they want to style the labels instead.